listenbrainz/webserver/static/js/tests/listens/ListenCountCard.test.tsx

```diff
@@ -1,15 +1,50 @@
 import * as React from "react";
 import { mount } from "enzyme";
 
+import GlobalAppContext, {
+  GlobalAppContextT,
+} from "../../src/utils/GlobalAppContext";
 import ListenCountCard from "../../src/listens/ListenCountCard";
+import APIService from "../../src/utils/APIService";
+
+const user = {
+  id: 1,
+  name: "track_listener",
+};
+
+const loggedInUser = {
+  id: 2,
+  name: "iliekcomputers",
+};
+
+const globalContext: GlobalAppContextT = {
+  APIService: new APIService("foo"),
+  currentUser: loggedInUser,
+};
 
 describe("ListenCountCard", () => {
   it("renders correctly when listen count is not zero", () => {
-    const wrapper = mount(<ListenCountCard listenCount={100} />);
+    const wrapper = mount(<ListenCountCard user={user} listenCount={100} />);
     expect(wrapper).toMatchSnapshot();
   });
   it("renders correctly when listen count is zero or undefined", () => {
-    const wrapper = mount(<ListenCountCard />);
+    const wrapper = mount(<ListenCountCard user={user} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+  it("renders user's name instead of 'You' when visiting another user's page", () => {
+    const wrapper = mount(
+      <GlobalAppContext.Provider value={globalContext}>
+        <ListenCountCard user={user} listenCount={100} />
+      </GlobalAppContext.Provider>
+    );
+    expect(wrapper).toMatchSnapshot();
```

Comment:
Snapshots are especially useful to make sure that the overall component renders as expected.
As you saw it's easy to update the snapshots, but harder to make sure nothing else in the snapshot file has changed that shouldn't have.

In this case, we want to test the rendered content directly in the test.
Instead of expect.toMatchSnapshot in the two new tests, we can ensure the name is rendered properly with the [enzyme library](https://enzymejs.github.io/enzyme/docs/api/) that we use to mount the components in the tests:

1. find the element by id
2. get the first child element
3. get the text content and compare it

Something like this: 
```
const outerCard = wrapper.find("#listen-count-card").first();
const innerDiv = outerCard.children().first();
expect(innerDiv.text()).toEqual("You have listened to 100 songs so far")
```

Reply:
P.S: the removal of `expect(wrapper).toMatchSnapshot();` in the last two tests means you'll also have to update the snapshots again.

Reply:
Hey! Thank you for the feedback. I understand what is to be done here.

Since the Listen count is surrounded by an `<hr />` and `<br />` respectively, they aren't considered as white space. So the expected text `"You have listened to 100 songs so far"` fails as it receives `"You have listened to100songs so far"` while running the test.

Code:
```html
<div class="card " id="listen-count-card">
    <div>You have listened to
        <hr />
        100
        <br />
        <small class="text-muted">songs so far</small>
    </div>
</div>
```

Is there any enzyme API that can be used for this? I have gone through their documentation, but no combination of their API is able to pass this test easily. All I can think apart from that is regex.

I do have a couple hacks to get around this -
1. Either test with `"You have listened to100songs so far"` to make it work. Probably not a good idea.
2. Or add white space before and after the Listen count in `ListenCountCard` component like so - 
```js
{` ${listenCount.toLocaleString()} `}
```

But again, these are hacks. We should write tests for the code and not the other way round.

I'd like to hear your thoughts on this.

Reply:
I've also tried using `.html()` over `.text()` to match the HTML directly, but it didn't look like an elegant solution.

Reply:
I think you are correct with the `.html()` solution, that was also my first thought.
I think that's elegant enough; in this case we can't easily test the text content, but we trust the HTML

Reply:
Alright, using `.html()` then :)

Reply:
@MonkeyDo Check out the new commit :)